backend/main/src/Group/Query/FindContainersOfAUserQuery/FindContainersOfAUserQuery.ts

```diff
@@ -0,0 +1,110 @@
+import { IQuery } from "src/Shared/Layers/Query/IQuery";
+import { NishikiDynamoDBClient } from "src/Shared/Adapters/DB/NishikiTableClient";
+import {
+	ContainerData,
+	UsersGroup,
+} from "src/Shared/Adapters/DB/NishikiDBTypes";
+import { Ok, Err, Result } from "result-ts-type";
+import { QueryError } from "src/Shared/Utils/Errors";
+import { isValidUUIDV4 } from "src/Shared/Utils/Validator";
+
+export class FindContainersOfAUserQuery
+	implements IQuery<{ userId: string }, IContainerData, InvalidUUId>
+{
+	constructor(private readonly nishikiDynamoDBClient: NishikiDynamoDBClient) {}
+
+	public async execute(input: { userId: string }): Promise<
+		Result<IContainerData, InvalidUUId>
+	> {
+		const { userId } = input;
+		if (!isValidUUIDV4(userId)) {
+			return Err(new InvalidUUId("User ID is invalid"));
+		}
+
+		const userGroups =
+			await this.nishikiDynamoDBClient.listOfUsersGroup(userId);
+
+		if (userGroups.length === 0) {
+			return Ok({ containers: [] });
+		}
+
+		const groupContainerDetailsList = (
+			await Promise.all(
+				userGroups.map(async (userGroup) => {
+					return await this.getGroupContainerDetails(userGroup);
+				}),
+			)
+		).flat();
```

Comment:
I'm not sure, but please tell me why you write like this. In my understanding, you can write it like this:
```suggestion
		cconst groupContainerDetailsList = (
			await Promise.all(
				userGroups.map((userGroup) => this.getGroupContainerDetails(userGroup)
			)
		)).flat();
```

Is my understanding incorrect?

Reply:
I'm so sorry. That's my fault. I should've removed the `async` after `.map`. I mean, at first, I directly wrote the content of `getGroupContainerDetails` in the map's callback argument place, so I needed to make the callback async function.